service/topology_coordinator.cc

```diff
@@ -1617,16 +1650,29 @@ class topology_coordinator : public endpoint_lifecycle_subscriber {
         co_await update_topology_state(std::move(guard), std::move(updates), "Finished tablet migration");
     }
 
-    future<> handle_tablet_resize_finalization(group0_guard g) {
+    future<group0_guard> generate_resize_finalization_updates(std::vector<canonical_mutation>& updates, group0_guard g) {
```

Comment:
should we rename it maybe_generate_resize_updates() or something? since it was now made conditional. then it's clearer on the caller that it might decide to not generate any if conditions aren't met.

Reply:
Done.